good & bad ending animations

PennyS8 opened the pull request.

PennyS8 committed:
imported assets

and duplicated a few of the carousel scripts to create icecream_truck scripts

PennyS8 committed:
truck scene

started copying the carousel scene to the truck scene

PennyS8 committed:
Icecream truck is in a scene

PennyS8 committed:
car go vroom

created an animation for the icecream truck driving

PennyS8 committed:
completed icecream truck driving animation

PennyS8 committed:
moved the truck

I put the ice-cream truck into a node then moved it in the scene so the animation player is using relative position to that node, instead of the scene.

PennyS8 committed:
created truck collision

PennyS8 committed:
Completed truck interaction stuffs

PennyS8 committed:
truck spawn timer = 10s

PennyS8 committed:
Good & bad ending animations

with music!

Next, add truck to the city park scene, join happyness meter to alternate endings, spooky sound effects and garage noise.

clarkm25 reviewed (approved):
Tested the ending scenes / ice cream truck. Everything ran well and code looks good.

clarkm25 committed:
Merge branch 'main' into Icecream-Truck-hand

clarkm25 committed:
Added player scene to the ending scene

- Added the actual player scene into ending scene in order to prevent it from crashing due to happiness not being instantiated.
